src/app/core/services/marios.service.ts

```diff
@@ -3,36 +3,76 @@ import { Injectable } from '@angular/core';
 import { Observable, ReplaySubject, Subject } from 'rxjs';
 import { takeUntil } from 'rxjs/operators';
 import { Marios } from '../models/marios.model';
-import { MariosList } from '../models/marios-list.model';
+import { MariosElement } from '../models/marios-element.model';
+import { SessionService } from './session.service';
 
 @Injectable({
   providedIn: 'root',
 })
 export class MariosService {
   private baseUrl = '/api/marios';
-  private mariosData: Marios[] = [];
-  private mariosList$ = new ReplaySubject<Marios[]>(1);
+  private lastMariosData: MariosElement[] = [];
+  private sentMariosData: MariosElement[] = [];
+  private receivedMariosData: MariosElement[] = [];
+
+  private lastMariosList$ = new ReplaySubject<MariosElement[]>(1);
+  private sentMariosList$ = new ReplaySubject<MariosElement[]>(1);
+  private receivedMariosList$ = new ReplaySubject<MariosElement[]>(1);
   private destroy$ = new Subject<void>();
 
-  constructor(private http: HttpClient) {}
+  private sentMariosListSize = 0;
+  private receivedMariosListSize = 0;
+
+  constructor(
+    private http: HttpClient,
+    private sessionService: SessionService
+  ) {}
+
+  get lastMariosList(): Observable<MariosElement[]> {
+    if (this.lastMariosData.length === 0) {
+      this.fetchMariosList();
+    }
+    return this.lastMariosList$.asObservable();
+  }
+
+  get sentMariosList(): Observable<MariosElement[]> {
+    if (this.sentMariosData.length === 0) {
+      this.fetchMariosList();
+    }
+    return this.sentMariosList$.asObservable();
+  }
 
-  get mariosList(): Observable<Marios[]> {
-    if (this.mariosData.length === 0) {
+  get receivedMariosList(): Observable<MariosElement[]> {
+    if (this.receivedMariosData.length === 0) {
       this.fetchMariosList();
     }
-    return this.mariosList$.asObservable();
+    return this.receivedMariosList$.asObservable();
+  }
+
+  public getSentMariosListSize(): number {
+    return this.sentMariosListSize;
+  }
+
+  public getReceiveMariosListSize(): number {
+    return this.receivedMariosListSize;
   }
 
   private fetchMariosList(): void {
     this.http
-      .get<Marios[]>(this.baseUrl)
+      .get<MariosElement[]>(this.baseUrl)
       .pipe(takeUntil(this.destroy$))
       .subscribe((data) => {
-        this.mariosData = [...data];
-        this.mariosList$.next(this.mariosData);
+        this.lastMariosData = [...data];
+        this.lastMariosList$.next(this.lastMariosData);
       });
   }
 
+  public refreshMariosList(): void {
+    this.fetchMariosList();
```

Comment:
Just for information: here you are calling functions that are below. refreshMariosList could be at then end, below these functions